crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -76,6 +76,8 @@ pub(crate) struct Checker<'a> {
     module_path: Option<&'a [String]>,
     /// Whether the current file is a stub (`.pyi`) file.
     is_stub: bool,
+    /// Whether the current file is a Jupyter notebook (`.ipynb`) file.
+    pub(crate) is_jupyter_notebook: bool,
```

Comment:
Nit: What do you think of passing in a `PythonFileType` enum instead of an enum to e.g. support Pyi files,. 

We could even go further. Rome has a `SourceType` struct that stores the `Language`, `version`, etc. Anything that may result in different parser behaviour

Reply:
would consider this more than a nit

Reply:
`JsFileSource` is what I meant. They must have renamed it. I think I prefer `PyFileType` or `PySourceType` to make it clear it's about the source file (which can be different from the codegen target, e.g. when transpiling Python 3.12 code to Python 3.8). 

> We might want to merge this with the existing SourceKind abstraction although that is used in a different context.

Oh, I didn't remember it existed. IT may be fine to have both until we split source files more clearly.